liquidctl/driver/lianli_uni.py

```diff
@@ -0,0 +1,105 @@
+from liquidctl.driver.usb import UsbHidDriver
+from liquidctl.error import NotSupportedByDevice
+
+
+_SYNC_ARGB_COMMAND_LENGTH = 7
+_SYNC_CHANNEL_COMMAND_LENGTH = 4
+
+_SYNC_ARGB_COMMANDS = {
+    "SL": [224, 16, 48, 0, 0, 0],
+    "AL": [224, 16, 65, 0, 0, 0],
+    "SL-Infinity": [224, 16, 97, 0, 0, 0],
+    "SL v2": [224, 16, 97, 0, 0, 0],
+    "AL v2": [224, 16, 97, 0, 0, 0],
+}
+_SYNC_CHANNEL_COMMANDS = {
+    "SL": [224, 16, 49],
+    "AL": [224, 16, 66],
+    "SL-Infinity": [224, 16, 98],
+    "SL v2": [224, 16, 98],
+    "AL v2": [224, 16, 98],
+}
+
+_PIDS = {
+    "SL": [0x7750, 0xA100],
+    "AL": [0xA101],
+    "SL-Infinity": [0xA102],
+    "SL v2": [0xA103, 0xA105],
+    "AL v2": [0xA104],
+}
+
+_CHANNEL_BYTE_MASK = 0x10
+_CHANNEL_PWM_MASK = 0x1
+
+
+class LianLiUNI(UsbHidDriver):
+    """Lian-Li Uni fans"""
+
+    _MATCHES = [
+        (0x0CF2, 0x7750, "LianLi-UNI SL", {"fan_count": 1, "temp_probs": 0, "led_channels": 2}),
+        (0x0CF2, 0xA100, "LianLi-UNI SL", {"fan_count": 1, "temp_probs": 0, "led_channels": 2}),
+        (0x0CF2, 0xA101, "LianLi-UNI AL", {"fan_count": 1, "temp_probs": 0, "led_channels": 2}),
+        (
+            0x0CF2,
+            0xA102,
+            "LianLi-UNI SL-Infinity",
+            {"fan_count": 4, "temp_probs": 0, "led_channels": 2},
+        ),
+        (0x0CF2, 0xA103, "LianLi-UNI SL v2", {"fan_count": 1, "temp_probs": 0, "led_channels": 2}),
+        (0x0CF2, 0xA105, "LianLi-UNI SL v2", {"fan_count": 1, "temp_probs": 0, "led_channels": 2}),
+        (0x0CF2, 0xA104, "LianLi-UNI AL v2", {"fan_count": 1, "temp_probs": 0, "led_channels": 2}),
+    ]
+
+    def __init__(self, device, description, fan_count, temp_probs, led_channels, **kwargs):
+        super().__init__(device, description, **kwargs)
+
+        self.variant = next(key for key in _PIDS if self.product_id in _PIDS[key])
+
+    def initialize(self, direct_access=False, fan_mode={}, **kwargs):
+        if 'rgb_sync' in kwargs and kwargs['rgb_sync']:
+            self._set_rgb_sync()
+
+    def set_fixed_speed(self, channel, duty, **kwargs):
+        if channel not in range(1, 7):
```

Comment:
`channel` here is `fan1`, `fan2` etc. from the command, in lowercase. You need to parse the actual channel number.

Reply:
Good catch, I just made a change.